webbpsf/distortion.py

```diff
@@ -147,9 +143,16 @@ def distort_image(hdulist_or_filename, ext=0, to_frame='sci', fill_value=0,
             assert xnew_coords.shape==ynew_coords.shape, "If new x and y inputs are a grid, must be same shapes"
             xnew, ynew = xnew_coords, ynew_coords
     elif to_frame=='sci':
-        xnew = xarr / oversamp + xnew_cen
-        ynew = yarr / oversamp + ynew_cen
+        osamp_x = aper.XSciScale / pixelscale
+        osamp_y = aper.YSciScale / pixelscale
+        osamp = (osamp_x + osamp_y) / 2
+        xnew = xarr / osamp + xnew_cen
+        ynew = yarr / osamp + ynew_cen
```

Comment:
Can you explain the logic here for averaging `osamp`? This computes first a distinct oversampling factor in each of X and Y, but then computes `xnew` and `ynew` using the average oversampling factor. Wouldn't it be more precise to calculate using the distinct oversampling factors per axis, like `xnew = xarr / osamp_x + xnew_cen`?

Reply:
I originally tried doing it per-axis, but `test_apply_distortion_pixel_scale()` would fail. I did not have time to look into why the test was failing, as I didn't quite understand the logic behind the test. Reverting to a single oversampling applied across both axes fixed the issue. Need to investigate further.

Reply:
Huh, yeah, now that I look at that test... I do not understand the logic either. 🤣